pkg/catalog/routes.go

```diff
@@ -435,3 +435,20 @@ func (mc *MeshCatalog) routesFromRules(rules []target.TrafficTargetRule, traffic
 
 	return routes, nil
 }
+
+// GetServicesForServiceAccounts returns a list of services corresponding to a list service accounts
+//	TODO: Consider merging this function and mc.GetServicesForServiceAccount in future (#2038)
+func (mc *MeshCatalog) GetServicesForServiceAccounts(saList []service.K8sServiceAccount) []service.MeshService {
+	serviceList := []service.MeshService{}
```

Comment:
Do you think we should have this as a map and then convert it to a slice to avoid duplicates ?

Reply:
yes good catch. will add logic to avoid duplicates.